tests/e2e/agent/test_get.py

```diff
@@ -77,7 +78,53 @@ def test_get_with_pdf_with_gemini_model(
         model=model,
     )
     assert isinstance(response, str)
-    assert "is a test page " in response.lower()
+    assert "is a test " in response.lower()
+
+
+@pytest.mark.parametrize(
+    "model",
+    [
+        ModelName.ASKUI__GEMINI__2_5__FLASH,
+        ModelName.ASKUI__GEMINI__2_5__PRO,
+    ],
+)
+def test_get_with_pdf_too_large(
+    vision_agent: VisionAgent,
+    model: str,
+    path_fixtures_dummy_pdf: pathlib.Path,
+    mocker: MockerFixture,
+) -> None:
+    mocker.patch(
+        "askui.models.askui.google_genai_api.MAX_FILE_SIZE_BYTES",
+        1,
+    )
+    with pytest.raises(ValueError, match="PDF file size exceeds the limit"):
+        vision_agent.get(
+            "What is in the PDF?",
+            source=path_fixtures_dummy_pdf,
+            model=model,
+        )
+
+
+def test_get_with_pdf_too_large_with_default_model(
+    vision_agent: VisionAgent,
+    path_fixtures_dummy_pdf: pathlib.Path,
+    mocker: MockerFixture,
+) -> None:
+    mocker.patch(
+        "askui.models.askui.google_genai_api.MAX_FILE_SIZE_BYTES",
+        1,
+    )
+
+    # This should raise a ValueError because the default model is Gemini and it falls
+    # back to inference askui which does not support pdfs
```

Comment:
Comment was very helpful. But I think we should change implementation to raise the also `PDF file size exceeds the limit` as this would be the reason for it not working.